datastore/providers/redis_datastore.py

```diff
@@ -293,54 +292,46 @@ async def _query(
         Takes in a list of queries with embeddings and filters and
         returns a list of query results with matching document chunks and scores.
         """
-        # Prepare results object
+        # Prepare query responses and results object
         results: List[QueryResult] = []
 
-        # Use asyncio for concurrent search
-        n = min(len(queries), 50)
-        semaphore = asyncio.Semaphore(n)
+        # Gather query results in a pipeline
+        logging.info(f"Gathering {len(queries)} query results", flush=True)
+        for query in queries:
 
-        async def _single_query(query: QueryWithEmbedding) -> QueryResult:
             logging.info(f"Query: {query.query}")
+            query_results: List[DocumentChunkWithScore] = []
+
             # Extract Redis query
             redis_query: RediSearchQuery = self._get_redis_query(query)
-            # Perform a single query
-            async with semaphore:
-                embedding = np.array(query.embedding, dtype=np.float64).tobytes()
-                # Get vector query response from Redis
-                query_response = await self.client.ft(REDIS_INDEX_NAME).search(
-                    redis_query, {"embedding": embedding}  # type: ignore
-                )
-                return query_response
-
-        # Concurrently gather query results
-        logging.info(f"Gathering {len(queries)} query results", flush=True)  # type: ignore
-        query_responses = await asyncio.gather(
-            *[_single_query(query) for query in queries]
-        )
+            embedding = np.array(query.embedding, dtype=np.float64).tobytes()
 
-        # Iterate through responses and construct results
-        for query, query_response in zip(queries, query_responses):
+            # Perform vector search
+            query_response = await self.client.ft(REDIS_INDEX_NAME).search(
+                redis_query, {"embedding": embedding}
+            )
 
-            # Iterate through nearest neighbor documents
-            query_results: List[DocumentChunkWithScore] = []
+            # Iterate through the most similar documents
             for doc in query_response.docs:
-                # Create a document chunk with score object with the result data
+                # Load JSON data
                 doc_json = json.loads(doc.json)
+                # Create document chunk object with score
                 result = DocumentChunkWithScore(
                     id=doc_json["metadata"]["document_id"],
                     score=doc.score,
                     text=doc_json["text"],
                     metadata=doc_json["metadata"],
+                    embedding=doc_json["embedding"]
```

Comment:
We don't actually want to return the embedding (this is too much data from a plugin for ChatGPT)